testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -359,21 +389,49 @@ fn run_case(
     }
 
     // Now validate using the stateless client if everything else passes
-    for (recovered_block, execution_witness) in &program_inputs {
+    for (recovered_block, execution_witnesses) in &program_inputs {
         let block = recovered_block.clone().into_block();
 
         // Recover the actual public keys from the transaction signatures
         let public_keys = recover_signers(block.body().transactions())
             .expect("Failed to recover public keys from transaction signatures");
 
-        stateless_validation_with_trie::<StatelessSparseTrie, _, _>(
-            block,
-            public_keys,
-            execution_witness.clone(),
+        // Validate stateless execution using a sparse trie for the storage access.
+        let (_, trie_post_state) = stateless_validation_with_trie::<StatelessSparseTrie, _, _>(
+            block.clone(),
+            public_keys.clone(),
+            execution_witnesses.trie.clone(),
             chain_spec.clone(),
             EthEvmConfig::new(chain_spec.clone()),
         )
         .expect("stateless validation failed");
+
+        // Validate stateless execution using a flatdb for the storage access.
+        let (_, flatdb_post_state) = stateless_validation_with_flatdb::<_, _>(
+            block.clone(),
+            public_keys.clone(),
+            execution_witnesses.flatdb.clone(),
+            chain_spec.clone(),
+            EthEvmConfig::new(chain_spec.clone()),
+        )
+        .expect("stateless validation with flatdb failed");
+
+        if trie_post_state != flatdb_post_state {
+            return Err(Error::Assertion(
+                "Post state mismatch between trie and flatdb implementations".to_string(),
+            ));
+        }
```

Comment:
We do a sanity check that both post-states of the normal execution+storage run, and the execution-only generated the same post-state. 

If that isn't the case, something is wrong.